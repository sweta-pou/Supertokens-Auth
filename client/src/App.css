.App {
  text-align: center;
  height: 100%;
  margin: 0;
}

body {
  display: flex;
  flex-direction: column;
  height: 100%;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

h1 {
  text-align: center;
}

.footer {
  background-color: rgb(0, 0, 0);
  bottom: 0;
  left: 0;
  padding-bottom: 0.6rem;
  padding-top: 0.6rem;
  position: fixed;
  width: 100%;
}

p {
  color: white;
  text-align: center;
}

.content {
  flex: 1;
  height: 120vh;
  overflow: auto;
}
