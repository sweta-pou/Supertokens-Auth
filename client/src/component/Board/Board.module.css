.square {
  background-color: black;
  display: grid;
  grid-template-columns: 6em 6em 6em;
  grid-template-rows: 6em 6em 6em;
  height: 18em;
  margin: 5em auto;
  padding: 0.0625em;
  width: 18em;
}
