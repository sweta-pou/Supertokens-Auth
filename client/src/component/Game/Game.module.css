.status {
  color: black;
  font-size: 1.5rem;
  text-align: center;
}

.list {
  list-style-position: inside;
  text-align: center;
}
