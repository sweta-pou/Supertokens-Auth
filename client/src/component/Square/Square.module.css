.squareComponent {
  background-color: rgba(255, 255, 255, 0.8);
  border: 0.0625em solid rgba(0, 0, 0, 0.8);
  font-size: 2rem;
  text-align: center;
}
