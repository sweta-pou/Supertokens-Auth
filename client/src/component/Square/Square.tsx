import React, { Component } from "react";
import classes from "./Square.module.css";

interface Props {
  value: string;
  onClicks(): void;
}

class Square extends Component<Props> {
  render() {
    return (
      <button
        className={classes.squareComponent}
        onClick={() => this.props.onClicks()}
      >
        {this.props.value}
      </button>
    );
  }
}

export default Square;
