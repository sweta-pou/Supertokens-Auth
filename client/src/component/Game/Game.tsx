import React, { Component } from "react";
import {
  signOut,
  redirectToAuth,
} from "supertokens-auth-react/recipe/emailpassword";
import Board from "../Board/Board";
import * as Winner from "../../winner/Winner";
import classes from "./Game.module.css";

interface IProps {}
interface IPartialStep {
  squares: Array<string>;
}
interface IState {
  history: Array<IPartialStep>;
  isXNext: Boolean;
  stepNumber: number;
}

class Game extends Component<IProps, IState, IPartialStep> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      history: [
        {
          squares: Array(9).fill(null),
        },
      ],
      isXNext: true,
      stepNumber: 0,
    };
  }
  clickHandler(i: number) {
    let history = this.state.history!.slice(0, this.state.stepNumber! + 1);
    let current = history[history.length - 1];
    let square = current.squares!.slice();
    if (Winner.calculateWinner(square) || square[i]) {
      return;
    }
    square[i] = this.state.isXNext ? "X" : "O";
    this.setState({
      history: history.concat([
        {
          squares: square,
        },
      ]),
      stepNumber: history.length,
      isXNext: !this.state.isXNext,
    });
  }
  jumpTo(n: number) {
    this.setState({
      stepNumber: n,
      isXNext: n % 2 === 0,
    });
  }
  Signout = async () => {
    await signOut();
    redirectToAuth();
  };
  render() {
    let history = this.state.history;
    let current = history![this.state.stepNumber!];
    let winner = Winner.calculateWinner(current.squares!);
    let status;
    if (winner) {
      status = "Winner : " + winner;
    } else {
      status = "Next player : " + (this.state.isXNext ? "X" : "O");
    }
    const moves = history!.map((step, move) => {
      const desc = move ? "Go to move #" + move : "Go to game start";
      return (
        <li key={move}>
          <button onClick={() => this.jumpTo(move)}>{desc}</button>
        </li>
      );
    });
    return (
      <>
        <button onClick={() => this.Signout()}>Sign Out</button>
        <p className={classes.status}>{status}</p>
        <Board
          squares={current.squares!}
          onClicked={(i: number) => this.clickHandler(i)}
        />
        <ol className={classes.list}>{moves}</ol>
      </>
    );
  }
}
export default Game;
