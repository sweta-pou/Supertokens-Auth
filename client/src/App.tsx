import React from "react";
import SuperTokens, {
  getSuperTokensRoutesForReactRouterDom,
} from "supertokens-auth-react";
import EmailPassword, {
  EmailPasswordAuth,
} from "supertokens-auth-react/recipe/emailpassword";
import Session from "supertokens-auth-react/recipe/session";
import { Switch, BrowserRouter as Router, Route } from "react-router-dom";
import "./App.css";
import Game from "./component/Game/Game";

interface Props {
  version: string;
}
SuperTokens.init({
  appInfo: {
    appName: "Tic-Tac-Toe",
    apiDomain: process.env.REACT_APP_API_apiDomain as string,
    websiteDomain: process.env.REACT_APP_API_websiteDomain as string,
  },
  recipeList: [EmailPassword.init(), Session.init()],
});
function App(props: Props) {
  return (
    <div>
      <h1>Tic-Tac-Toe</h1>
      <div className="content">
        <Router>
          <Switch>
            {getSuperTokensRoutesForReactRouterDom(require("react-router-dom"))}
            <Route path="/">
              <EmailPasswordAuth>
                <Game />
              </EmailPasswordAuth>
            </Route>
          </Switch>
        </Router>
      </div>
      <div className="footer">
        <p>v {props.version}</p>
      </div>
    </div>
  );
}

export default App;
