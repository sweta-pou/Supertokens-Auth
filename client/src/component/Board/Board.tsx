import React, { Component } from "react";
import Square from "../Square/Square";
import classes from "./Board.module.css";

interface Props {
  squares: Array<string>;
  onClicked(i: number): void;
}

class Board extends Component<Props> {
  render() {
    return (
      <>
        <div className={classes.square}>
          {[...Array(9)].map((x, i) => (
            <Square
              key={i}
              value={this.props.squares![i]}
              onClicks={() => this.props.onClicked(i)}
            />
          ))}
        </div>
      </>
    );
  }
}

export default Board;
