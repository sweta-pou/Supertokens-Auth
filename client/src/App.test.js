import { render, screen } from "@testing-library/react";
import App from "./App";
import Game from "./component/Game/Game";
import Board from "./component/Board/Board";
import Square from "./component/Square/Square";
import renderer from "react-test-renderer";

test("renders learn react link", () => {
  render(<App />);
  const linkElement = screen.getByText(/Tic-Tac-Toe/i);
  expect(linkElement).toBeInTheDocument();
});

it("renders App correctly", () => {
  const tree = renderer.create(<App version={"0.1.0 + 1"} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Game correctly", () => {
  const tree = renderer.create(<Game />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Board correctly", () => {
  const tree = renderer
    .create(
      <Board squares={["X", "O", null, null, "X", null, null, null, null]} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Square correctly", () => {
  const tree = renderer
    .create(<Square value={["X", "X", "O", "O", "X", "X", "O", "X", "O"]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
