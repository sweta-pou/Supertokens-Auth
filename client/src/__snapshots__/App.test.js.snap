// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<div>
  <h1>
    Tic-Tac-Toe
  </h1>
  <div
    className="content"
  />
  <div
    className="footer"
  >
    <p>
      v 
      0.1.0 + 1
    </p>
  </div>
</div>
`;

exports[`renders Board correctly 1`] = `
<div
  className="square"
>
  <button
    className="squareComponent"
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="squareComponent"
    onClick={[Function]}
  >
    O
  </button>
  <button
    className="squareComponent"
    onClick={[Function]}
  />
  <button
    className="squareComponent"
    onClick={[Function]}
  />
  <button
    className="squareComponent"
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="squareComponent"
    onClick={[Function]}
  />
  <button
    className="squareComponent"
    onClick={[Function]}
  />
  <button
    className="squareComponent"
    onClick={[Function]}
  />
  <button
    className="squareComponent"
    onClick={[Function]}
  />
</div>
`;

exports[`renders Game correctly 1`] = `
Array [
  <button
    onClick={[Function]}
  >
    Sign Out
  </button>,
  <p
    className="status"
  >
    Next player : X
  </p>,
  <div
    className="square"
  >
    <button
      className="squareComponent"
      onClick={[Function]}
    />
    <button
      className="squareComponent"
      onClick={[Function]}
    />
    <button
      className="squareComponent"
      onClick={[Function]}
    />
    <button
      className="squareComponent"
      onClick={[Function]}
    />
    <button
      className="squareComponent"
      onClick={[Function]}
    />
    <button
      className="squareComponent"
      onClick={[Function]}
    />
    <button
      className="squareComponent"
      onClick={[Function]}
    />
    <button
      className="squareComponent"
      onClick={[Function]}
    />
    <button
      className="squareComponent"
      onClick={[Function]}
    />
  </div>,
  <ol
    className="list"
  >
    <li>
      <button
        onClick={[Function]}
      >
        Go to game start
      </button>
    </li>
  </ol>,
]
`;

exports[`renders Square correctly 1`] = `
<button
  className="squareComponent"
  onClick={[Function]}
>
  X
  X
  O
  O
  X
  X
  O
  X
  O
</button>
`;
